calcom/cal.com
# fix: don't send request reminder for unpaid bookings

## What does this PR do?

This prevents "Awaiting Approval" emails from wrongly being sent out for unpaid bookings.

Fixes #10253

It might make sense to create a new "Unpaid Booking" reminder email to send out to Organizers so they can take action, e.g. reject/cancel unpaid bookings. But given the proposed changes to payment flow in #6809, this may be premature.

## Requirement/Documentation

N/A

## Type of change


## How should this be tested?

Create 4 bookings:

- A. paid event -> don't pay
- B. paid event with opt-in -> pay
- C. paid event with opt-in -> don't pay
- D. free event with opt-in

Reduce create dates in DB by 3h, then trigger reminder job via:

```sh
APP_URL=localhost:3000
CRON_API_KEY=0cc0e6c35519bba620c9360cfe3e68d0

curl "${APP_URL}/api/cron/bookingReminder" \
  -X POST \
  -H "content-type: application/json" \
  -H "authorization: ${CRON_API_KEY}"
```
See that reminders are only sent for bookings B and D.

## Mandatory Tasks
